python/pysmurf/client/base/base_class.py

```diff
@@ -85,7 +85,8 @@ def __init__(self, log=None, epics_root=None, offline=False,
         # do this than just hardcoding paths? This needs to be cleaned
         # up somehow
 
-        self.epics_root = epics_root
+        if epics_root is not None:
```

Comment:
Do we need a default value for this? What do we want to happen if `epics_root` is None?

Reply:
Added a comment about how default is taken from cfg file by the preceeding `self.copy_config_to_properties` call (since that's super not obvious) and fixed my logic error so that `epics_root` can't accidentally end up set to None (whoops):
https://github.com/slaclab/pysmurf/pull/454/commits/9d9883ae69f6f6ab855a4a3052ac0901dedd61c1
also changed logic for offline slightly ; now just setting to '' in offline mode even if user provides an `epics_root`.  @eyyoung24 lemme know if that breaks something.